### ISSUE TITLE
Add classification metrics: `SensitivityAtSpecificity` and `SpecificityAtSensitivity`

### ISSUE THREAD

**Issue by iamharsha1999** [Lightning-AI/metrics#971]
### Discussed in https://github.com/PyTorchLightning/metrics/discussions/965

<div type='discussions-op-text'>

<sup>Originally posted by **iamharsha1999** April 19, 2022</sup>
I was specifically looking for the following two metrics:
1. Sensitivity at Specificity
2. Specificity at Sensitivity

However I was not able to find it ? Would it be possible to add these implementations to torchmetrics. 

You can find the tensorflow implementations below:
https://www.tensorflow.org/api_docs/python/tf/keras/metrics/SpecificityAtSensitivity
https://www.tensorflow.org/api_docs/python/tf/keras/metrics/SensitivityAtSpecificity

**Comment by SkafteNicki** [comment]
Hi @iamharsha1999,
You are completely right that these metrics are not implemented in torchmetrics, yet ;)
Would you like to contribute them to the project?

**Comment by angadkalra** [comment]
Any progress on this? I actually implemented this metric for my own project and can contribute if needed.

**Comment by stancld** [comment]
Hi @angadkalra, do you have any updates here? :]

**Comment by angadkalra** [comment]
Sorry for the late reply! I will contribute ASAP.

**Comment by iamharsha1999** [comment]
@angadkalra @SkafteNicki Hey any update on this issue. I can try this out and contribute.

**Comment by angadkalra** [comment]
@iamharsha1999 go ahead, I'm busy with work as well.

**Comment by shenoynikhil** [comment]
I had a draft local branch on this a while back. Just put it up since there had been no progress. Feel free to implement the other variant `SensitivityAtSpecificity` if you guys are interested.
